vivisect/__init__.py

```diff
@@ -1950,10 +1962,42 @@ def makeString(self, va, size=None):
         if size <= 0:
             raise Exception("Invalid String Size: %d" % size)
 
+        # rip through the desired memory range to populate any substrings
+        subs = set()
```

Comment:
does this cover both use-cases listed above?  (1)little string discovered first, (2)big string discovered first?
please document with some comments how it works.  this is wall-o-code-time.

i think i like the substrings concept, but i haven't seen code which effectively removes previously discovered strings, if that's a thing.  do we end up with multiple string locations within each other (i'm not completely opposed to that approach, just need to understand and don't have too many mental cycles to parse through this).

Reply:
Aye, added some docs.

And it should cover both cases. I even have some tests for it :)